homeassistant/components/bittrex/__init__.py

```diff
@@ -0,0 +1,228 @@
+                    currency = "USDT-EUR"
+
+                if currency not in asset_tickers_dict:
+                    asset_tickers_dict[currency] = {}
+                    ticker_details = tickers[currency]
+                    asset_tickers_dict[currency].update(ticker_details)
+
+        return asset_tickers_dict
+
+    @staticmethod
+    def _prep_balances(balances, tickers):
+        """Prepare balances data."""
+
+        balances_dict = {}
+
+        for balance in balances:
+            if (
+                balances[balance]["currencySymbol"] not in balances_dict
+                and balances[balance]["currencySymbol"] != "BTXCRD"
+            ):
+                balances_dict[balances[balance]["currencySymbol"]] = {}
+                balances_dict[balances[balance]["currencySymbol"]].update(
+                    balances[balance]
+                )
+                base_asset_ticker_details = None
+
+                total_balance = float(balances[balance]["total"])
+
+                if ASSET_VALUE_BASE not in balances[balance]["currencySymbol"]:
+                    # Prevent that we try to search ASSET_VALUE_BASE+ASSET_VALUE_BASE (e.g. BTCBTC)
+                    base_asset_symbol = (
+                        f"{balances[balance]['currencySymbol']}-{ASSET_VALUE_BASE}"
+                    )
+
+                    # Flip currency format as it differs from other currencies
+                    if base_asset_symbol == "EUR-USDT":
+                        base_asset_symbol = "USDT-EUR"
+
+                    base_asset_ticker_details = tickers[base_asset_symbol]
+
+                else:
+                    balances_dict[balances[balance]["currencySymbol"]][
+                        "asset_value_in_base_asset"
+                    ] = total_balance
+
+                if base_asset_ticker_details:
+                    # If we can find a pair with ASSET_VALUE_BASE, include it in the dict
+                    balances_dict[balances[balance]["currencySymbol"]][
+                        ASSET_VALUE_BASE
+                    ] = {}
+                    balances_dict[balances[balance]["currencySymbol"]][
+                        ASSET_VALUE_BASE
+                    ].update(base_asset_ticker_details)
+                    balances_dict[balances[balance]["currencySymbol"]][
+                        "asset_value_in_base_asset"
+                    ] = total_balance * float(
+                        base_asset_ticker_details["lastTradeRate"]
+                    )
+
+        return balances_dict
+
+    async def _async_update_data(self):
+        """Fetch Bittrex data."""
+        try:
+            bittrex = Bittrex(self._api_key, self._api_secret)
+
+            tickers = await bittrex.get_tickers()
+            markets = await bittrex.get_markets()
+            balances = await bittrex.get_balances()
+            open_orders = await bittrex.get_open_orders()
+            closed_orders = await bittrex.get_closed_orders()
```

Comment:
It's possible to run all these coroutines concurrently, as tasks using `asyncio.gather`. This can speed up the process because you are spawning 5 tasks at the same time instead of awaiting these commands one-by-one.

Something like this:
```suggestion
        tickers, markets, balances, open_orders, closed_orders = await asyncio.gather(
            bittrex.get_tickers(),
            bittrex.get_markets(),
            bittrex.get_balances(),
            bittrex.get_open_orders(),
            bittrex.get_closed_orders(),
        )
```

Reply:
Thanks - added your suggestion.